rubix-spi/src/main/java/com/qubole/rubix/spi/ClusterManager.java

```diff
@@ -46,6 +49,11 @@
  */
 public abstract class ClusterManager
 {
+  private static Log log = LogFactory.getLog(ClusterManager.class);
+
+  private int currentNodeIndex = -1;
+  private String nodeHostname;
```

Comment:
should we initialize to empty string to avoid any null failures?

Reply:
Actually we should throw exception in hostname/address initialisation itself otherwise we will hit NPE on nodeCache later. Will add a commit for that. Will add requireNonNull on accessors of these variables